src/apps/vpn/models/recommendedlocationmodel.cpp

```diff
@@ -54,12 +58,21 @@ void RecommendedLocationModel::refreshModel() {
   QList<const ServerCity*> cities = recommendedLocations(DEFAULT_ENTRIES);
   if (m_recommendedCities.length() != cities.length()) {
     beginResetModel();
-    m_recommendedCities.swap(cities);
+    m_recommendedCities.clear();
+    for (auto& city : cities) {
+      // Wrap that pointers into a
+      // QPointer, so that we are notified if they are
+      // deleted.
+      m_recommendedCities.append(QPointer((ServerCity*)city));
```

Comment:
Is it possible to change the return type of recommendedLocations to QList<QPointer<ServerCity>>? If so, that would be a cleaner approach?

Reply:
Then how about introducing RecommendedLocationModel::recommendedLocationsRaw() for calling from QML, which would call RecommendedLocationModel::recommendedLocations and convert the returned QList<QPointer<ServerCity>> to QList<const ServerCity*>? Then C++ code would call recommendedLocations() and QML would call recommendedLocationsRaw()